### ISSUE TITLE
Redundant counter in batchnorm impl

### ISSUE THREAD

**Issue by soumyarooproy** [pytorch/pytorch#30440]
## 🐛 Bug

This looks like an efficiency/perf bug (unless I am missing something).

`self.num_batches_tracked` is initialized and incremented but never used when `self.momentum` is not `None` during training (and tracking running stats).

## To Reproduce

Review code:
https://github.com/pytorch/pytorch/blob/b8f50d9cc860460c7e9b6d3b370cab546e9f9583/torch/nn/modules/batchnorm.py#L35

https://github.com/pytorch/pytorch/blob/b8f50d9cc860460c7e9b6d3b370cab546e9f9583/torch/nn/modules/batchnorm.py#L97-L102

## Expected behavior

`self.num_batches_tracked` should be set to `None` when `self.momentum` is not `None` to achieve a cleaner implementation

## Environment

NA

## Additional context

NA

**Comment by yf225** [comment]
The `num_batches_tracked` attribute of `BatchNorm{1,2,3}d` is part of the public API, and removing it would break backward compatibility. I would say this is probably a won't fix, unless we prove that it significantly affects perf.

**Comment by soumyarooproy** [comment]
Hi!

I am not suggesting removing it entirely. Just suggesting cleaning up its def based on its use.

The current implementation indicates that this attribute takes as values `None` or a 0-d `long` tensor and this will continue to be the case after the clean-up, no? Will that still be a breaking change?
